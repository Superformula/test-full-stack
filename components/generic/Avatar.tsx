import { ReactElement, ImgHTMLAttributes } from 'react'

import styles from './Avatar.module.css'

export default function Avatar({
  src,
  width = '126',
  height = '126',
  className,
  ...rest
}: ImgHTMLAttributes<HTMLImageElement>): ReactElement {
  return (
    <img
      src={src}
      width={width}
      height={height}
      className={`${styles.avatar} ${className}`}
      {...rest}
    />
  )
}
