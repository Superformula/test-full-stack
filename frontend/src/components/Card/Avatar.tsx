import styles from "./Card.module.css";
import { useEffect, useState } from "react";
import { createApi } from "unsplash-js";

const api = createApi({
  // Don't forget to set your access token here!
  // See https://unsplash.com/developers
  accessKey: "M1dsgksncu-N8pjPvgshjGXU6tRjhWjpqIfD28rtyg0",
});

const Avatar = () => {
  const [data, setPhotosResponse] = useState(null);

  useEffect(() => {
    api.photos
      .getRandom({ query: "portrait", orientation: "portrait" })
      .then((result) => {
        setPhotosResponse(result);
      })
      .catch(() => {
        console.log("something went wrong!");
      });
  }, []);

  console.log(data);

  if (!data) {
    return null;
  }

  return <img className={styles.avatar} src={data.response.urls.regular} />;
};

export default Avatar;
