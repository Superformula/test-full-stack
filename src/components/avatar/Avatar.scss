.avatar {
  height: 168px;
  width: 168px;
  border-radius: 50%;
  object-fit: cover;
}
