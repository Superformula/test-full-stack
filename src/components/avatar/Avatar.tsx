import classnames from 'classnames';
import { memo } from 'react';

import classes from './Avatar.module.scss';

interface AvatarProps {
  src: string;
  title: string;
  className?: string;
}

function AvatarComponent(props: AvatarProps) {
  const imageUrl = props.src ? props.src : 'assets/profile.jpg';
  return (
    <div
      className={classnames(classes.avatar, props.className)}
      style={{
        backgroundImage: `url("${imageUrl}")`,
      }}
    />
  );
}

export const Avatar = memo(AvatarComponent);
