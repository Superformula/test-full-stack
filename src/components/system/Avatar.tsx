import styled from "@emotion/styled";

export const Avatar = styled.div`
  background-color: ${({ theme }) => theme.color.secondary};
  border-radius: 50%;
  height: 10.5rem;
  width: 10.5rem;
`;
