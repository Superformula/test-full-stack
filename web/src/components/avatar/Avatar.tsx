import React, { ReactElement } from 'react'
import style from './Avatar.module.scss'

interface AvatarProps {
  src: string
  alt: string
}

export const Avatar = ({ src, alt }: AvatarProps): ReactElement => (
  <div className={style.avatar}>
    <img src={src} alt={alt} />
  </div>
)
