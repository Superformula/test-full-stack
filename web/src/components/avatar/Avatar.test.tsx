import { render } from '@testing-library/react'
import React from 'react'
import { Avatar } from './Avatar'

test('renders avatar and matches snapshot', () => {
  const { container } = render(<Avatar src="https://image.com/photo.png" alt="Test picture" />)
  expect(container).toMatchSnapshot()
})
