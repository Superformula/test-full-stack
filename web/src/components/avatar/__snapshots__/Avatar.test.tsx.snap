// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders avatar and matches snapshot 1`] = `
<div>
  <div
    class="avatar"
  >
    <img
      alt="Test picture"
      src="https://image.com/photo.png"
    />
  </div>
</div>
`;
